GitUp/Application/AppDelegate.m

```diff
@@ -261,19 +295,6 @@ - (void)_showNotificationWithTitle:(NSString*)title action:(SEL)action message:(
   [[NSUserNotificationCenter defaultUserNotificationCenter] deliverNotification:notification];
 }
 
-- (void)handleDocumentCountChanged {
-  BOOL showWelcomeWindow = [[NSUserDefaults standardUserDefaults] boolForKey:kUserDefaultsKey_ShowWelcomeWindow];
-  if (showWelcomeWindow && (_allowWelcome > 0) && ![[[NSDocumentController sharedDocumentController] documents] count]) {
-    if (!_welcomeWindow.visible) {
-      [_welcomeWindow makeKeyAndOrderFront:nil];
-    }
-  } else {
-    if (_welcomeWindow.visible) {
-      [_welcomeWindow orderOut:nil];
-    }
-  }
-}
-
```

Comment:
Seems like we should still control the window being displayed from here if we're going to keep the logic equivalent.

Reply:
@lucasderraugh ... and goes to welcome window controller 😄 
If we want to add additional logic, which is not related to welcome window lifecycle, we could expand this method in the same way by forwarding it to appropriate controller.